src/coprocessor/codec/mysql/time/mod.rs

```diff
@@ -811,25 +809,25 @@ impl Display for Time {
     }
 }
 
-impl<T: Write> TimeEncoder for T {}
+impl<T: NumberEncoder> TimeEncoder for T {}
 
 /// Time Encoder for Chunk format
 pub trait TimeEncoder: NumberEncoder {
     fn encode_time(&mut self, v: &Time) -> Result<()> {
         use num::ToPrimitive;
 
         if !v.is_zero() {
-            self.encode_u16(v.time.year() as u16)?;
+            self.write_u16(v.time.year() as u16)?;
             self.write_u8(v.time.month() as u8)?;
             self.write_u8(v.time.day() as u8)?;
             self.write_u8(v.time.hour() as u8)?;
             self.write_u8(v.time.minute() as u8)?;
             self.write_u8(v.time.second() as u8)?;
-            self.encode_u32(v.time.nanosecond() / 1000)?;
+            self.write_u32(v.time.nanosecond() / 1000)?;
         } else {
             let len = mem::size_of::<u16>() + mem::size_of::<u32>() + 5;
             let buf = vec![0; len];
```

Comment:
Let's change it to `[0; len]` instead of `vec![0; len]` to avoid dynamic allocation

Reply:
if we use array , the `len` must be known at compile time, so we change it to 
```
  let buf = vec![0; 11]; // mem::size_of::<u16>() + mem::size_of::<u32>() + 5 == 11
```
?

Reply:
`mem::size_of()` is actually a constant known at compile time.
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=a95bd5bb36a0f0cbab4b3823e936bbab

Reply:
For this one you may simply change `let len = ...` to `const LEN: usize = ...`. Then you can use it in `[0; LEN]`.

Reply:
oh, thanks ~